### PR TITLE
[stdlib] Isolate Array bridging behavior for ABI stability

### PR DESCRIPTION
**Pull request by lancep**
rdar://problem/35943342

### CODE REVIEW

**Comment by lancep** [comment]
@swift-ci please smoke benchmark

**Comment by lancep** [comment]
@swift-ci please test

**Comment by lancep** [comment]
@swift-ci please test

**Comment by lancep** [comment]
I think this is good to merge but I'd like some extra eyes on it first

**Comment by lancep** [comment]
@swift-ci please smoke benchmark

File: stdlib/public/core/Arrays.swift.gyb [hunk 1]
```diff
@@ -2381,6 +2381,11 @@ extension ${Self} {
 %end
 
 #if _runtime(_ObjC)
+@_versioned internal
```
Comment (milseman) [inline stdlib/public/core/Arrays.swift.gyb]: Please add a comment describing why this is here. Otherwise, someone will just delete it

**Comment by lancep** [comment]
@swift-ci please smoke test

**Review by airspeedswift — APPROVED** [review]
with that comment, lgtm.

**Comment by milseman** [comment]
Did you have an eager bridging PR? We could check in something that conditionally enables it based on build parameters.

**Comment by lancep** [comment]
I do, but it'll need to be substantially changed if we were to do that.

**Pull request merged**